DllImportGenerator/DllImportGenerator/Marshalling/StringMarshaller.Ansi.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+using static Microsoft.Interop.MarshallerHelpers;
+
+namespace Microsoft.Interop
+{
+    internal sealed class AnsiStringMarshaller : ConditionalStackallocMarshallingGenerator
+    {
+        private static readonly TypeSyntax NativeType = PointerType(PredefinedType(Token(SyntaxKind.ByteKeyword)));
+
+        private readonly IMarshallingGenerator utf8StringMarshaller;
+
+        public AnsiStringMarshaller(IMarshallingGenerator utf8StringMarshaller)
```

Comment:
Why passing in the UTF8 marshaller? If we need to name it instead it might make sense to pass in the actual class and not the interface.